packages/react-components/src/molecules/LabeledMultiSelect.tsx

```diff
@@ -16,15 +17,40 @@ type LabeledMultiSelectProps = {
   readonly title: ReactNode;
   readonly subtitle?: React.ReactNode;
   readonly description?: ReactNode;
-} & Exclude<ComponentProps<typeof MultiSelect>, 'id'>;
-const LabeledMultiSelect: React.FC<LabeledMultiSelectProps> = ({
+};
+export type SyncLabeledMultiSelectProps = {
```

Comment:
We can add a helper method `isSync(data: Sync... | Async...): data is Sync { return data?.async === false }` and use that in the if in `forContent`. Then we can avoid the type casts (I think)

Reply:
We are going to have another PR to refactor this 👍 
cc @kmaid